hs_swat_modelinstance/receivers.py

```diff
@@ -1,27 +1,48 @@
 from django.dispatch import receiver
+from django.core.exceptions import ObjectDoesNotExist
 
-from hs_core.signals import pre_metadata_element_create, pre_metadata_element_update,pre_create_resource
+from hs_core.signals import pre_metadata_element_create, pre_metadata_element_update,\
+    pre_create_resource, post_metadata_element_update
 
-from hs_swat_modelinstance.models import SWATModelInstanceResource
+import hs_swat_modelinstance.models as swat_models
 from hs_swat_modelinstance.forms import ModelOutputValidationForm, ExecutedByValidationForm,\
     ModelObjectiveValidationForm, SimulationTypeValidationForm, ModelMethodValidationForm,\
     ModelParameterValidationForm, ModelInputValidationForm
 
 
-@receiver(pre_create_resource, sender=SWATModelInstanceResource)
+@receiver(pre_create_resource, sender=swat_models.SWATModelInstanceResource)
 def swatmodelinstance_pre_create_resource(sender, **kwargs):
     metadata = kwargs['metadata']
     modeloutput = {'modeloutput': {'includes_output': False}}
     metadata.append(modeloutput)
 
-@receiver(pre_metadata_element_create, sender=SWATModelInstanceResource)
+
+@receiver(pre_metadata_element_create, sender=swat_models.SWATModelInstanceResource)
 def metadata_element_pre_create_handler(sender, **kwargs):
     return _process_metadata_update_create(**kwargs)
 
-@receiver(pre_metadata_element_update, sender=SWATModelInstanceResource)
+
+@receiver(pre_metadata_element_update, sender=swat_models.SWATModelInstanceResource)
 def metadata_element_pre_update_handler(sender, **kwargs):
     return _process_metadata_update_create(**kwargs)
 
+
+@receiver(post_metadata_element_update, sender=swat_models.SWATModelInstanceResource)
+def check_element_exist(sender, **kwargs):
```

Comment:
Here is an example of testing a view function. The testing of the view function should cover test for the function in the receivers:

https://github.com/hydroshare/hydroshare/blob/develop/hs_app_timeseries/tests/test_update_sqlite_file_view_function.py

In order to test your receiver function you have to write a unit test for the view function in he_core that updates metadata. For now if you like create an issue to do it after this PR.